rfcs/20230623-compatibility.md

```diff
@@ -0,0 +1,95 @@
+# [RFC] Increase StableHLO Compatibility Guarantees
+
+Status: Approved<br/>
+Initial version: 6/23/2023<br/>
+Last updated: 4/30/2024<br/>
+Discussion thread: [openxla-discuss][openxla-discuss-post].
+
+## Summary
+
+**Proposal:** For StableHLO v1.0 Release, provide 5yrs backward compatibility,
+2yrs forward compatibility.
+
+ML Models deployed on-device (eg., Android) need strict backward and forward
+compatibility guarantees.
+
+* A deployed ML Model should never break due to a software update. This could
+  be an update to the ML runtime, Mobile OS, or the App itself. OEMs regularly
+  update phones, which can break functionality if the Opset changes.
+* ML models are often long-lived. Even when the application is updated, the
+  model it uses may be older or the application team may not have access to
+  the source model it uses. Said differently, a mobile ML runtime needs to
+  support older versions of StableHLO Ops.
+* There are a significant number of users who use old Mobile/Android phones,
+  often **5+ years**. App developers should be able to target older phones,
+  should they choose to for deploying their ML features.
+* Release cycles for new on-device toolkits are roughly annual, so a
+  **2yr forward compatiblity** window is needed to target the latest release. We
+  see this requirement for other users in the StableHLO community on annual
+  release cycles.
+
+Due to the above it's essential that Opset definitions are maintained long-term.
+It's reasonable for us as a community to iterate on the Opset and utilize the
+VHLO mechanism to version them. But once we have ML models deployed in the
+market, especially on Mobile phones it's not feasible to update the execution
+environment.
+
+## OpenXLA Discuss Summary
+
+_For full discussion see the [OpenXLA Discuss post][openxla-discuss-post]._
+
+### Sep 18, 2023
+
+As mentioned in TensorFlow Lite update on StableHLO use, TFLite would like to
+consume StableHLO as an input source and leverage VHLO versioning facilities. In
+order to achieve this goal, we propose the following changes to the StableHLO
+compatibility - please let us know of any feedback you have!
+
+#### Proposed Compatibility Window
+
+On average, Android phones receive 3 years of OS update guarantee(e.g. Pixel
+update Policy, update policy from other companies), and such OS updates must not
+break existing models on-device. Given that, we would like to propose a backward
+compatibility window of 3 years for StableHLO to support this use case.
+Additionally, new models that don’t use new features must be runnable on these
+supported OS, so we would also like to propose a 3 year forward compatibility
+window. And we would like to work with you to determine when should the
+compatibility window be extended.
```

Comment:
These numbers are different from the ones listed at the very top of the RFC. Is that intentional? What am I missing?

Reply:
The proposal evolved over time. I removed this section all together as it doesn't add anything in its current state.